src/door.rs

```diff
@@ -0,0 +1,291 @@
+use super::cfg::{Config, Door, User};
+use super::dos::Templates;
+use super::{LaunchArgs, SysopCmdArgs};
+
+use chrono::Local;
+use fs4::FileExt;
+use serde::Serialize;
+use std::env;
+use std::fs;
+use std::path::Path;
+use std::process::{Child, Command, Stdio};
+
+#[derive(Serialize, Debug)]
+struct LaunchVars<'a> {
+    user: &'a User,
+    node: i8,
+    current_time: String,
+}
+
+#[derive(Serialize, Debug)]
+struct BatchCommands {
+    commands: String,
+}
+
+fn get_door<'a>(name: &String, config: &'a Config) -> &'a Door {
+    let door = match config.doors.get(name) {
+        Some(door) => door,
+        None => {
+            eprintln!("Unknown door: {0}", name);
+            std::process::exit(1);
```

Comment:
See previous comments about errors. If you want to keep the error management in the function, you could try using `expect`, it will panic with the given error message. Panicking exits with status code 101.